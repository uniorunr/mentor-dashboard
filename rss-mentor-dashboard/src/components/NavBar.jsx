import React from 'react';

function App({ greet }) {
  return (
    <div>
      <h1>{greet}</h1>
    </div>
  );
}

export default App;
