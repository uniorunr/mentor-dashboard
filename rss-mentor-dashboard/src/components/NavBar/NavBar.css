.navbar-container {
  display: flex;
  justify-content: center;
}

.navbar-container .navbar {
  width: 90%;
  margin: 1% 0;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}
