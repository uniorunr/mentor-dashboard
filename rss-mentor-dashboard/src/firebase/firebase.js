import * as firebase from 'firebase/app';
import 'firebase/auth';

const API_KEY = 'AIzaSyCum54wsr0slpXjQFvIToTEUUlCDkf0RuI';
const AUTH_DOMAIN = 'rss-mentor-dashboard-8a0db.firebaseapp.com';
const DATABASE_URL = 'https://rss-mentor-dashboard-8a0db.firebaseio.com';
const PROJECT_ID = 'rss-mentor-dashboard-8a0db';
const STORAGE_BUCKET = 'rss-mentor-dashboard-8a0db.appspot.com';
const MESSAGING_SENDER_ID = '917025581702';

class FireBase {
  static init() {
    const config = {
      apiKey: API_KEY,
      authDomain: AUTH_DOMAIN,
      databaseURL: DATABASE_URL,
      projectId: PROJECT_ID,
      storageBucket: STORAGE_BUCKET,
      messagingSenderId: MESSAGING_SENDER_ID,
    };
    firebase.initializeApp(config);
  }

  static async auth() {
    const token = await firebase.auth().signInWithPopup(new firebase.auth.GithubAuthProvider())
      .then(result => result.credential.accessToken)
      .catch((error) => {
        throw new Error(error);
      });

    const userData = await fetch(`https://api.github.com/user?access_token=${token}`)
      .then(response => response.json())
      .catch((error) => {
        throw new Error(error);
      });

    const user = firebase.auth().currentUser;

    localStorage.setItem('userStatus', 'loggedIn');

    user.updateProfile({
      displayName: userData.login,
      photoURL: userData.avatar_url,
    }).catch((error) => {
      throw new Error(error);
    });
  }

  static logout() {
    firebase.auth().signOut().then(() => {
      window.location.reload();
    }).catch((error) => {
      throw new Error(error);
    });
    localStorage.removeItem('selectedMentor');
  }
}

export default FireBase;
