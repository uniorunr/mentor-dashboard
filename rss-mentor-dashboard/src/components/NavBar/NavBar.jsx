import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from './Logo/Logo';
import UserPicker from './UserPicker/UserPicker';
import './NavBar.css';
import LoginButton from './Login/Login';
import UserInfo from './UserInfo/UserInfo';
import { mentorsList } from '../../utils/parseJSON';

class NavBar extends Component {
  render() {
    const { handleInput, mentorDataObj } = this.props;
    return (
      <div className="navbar-container">
        <nav className="navbar">
          <Logo link="/" brandName="RSS MD" />
          <UserPicker
            placeholder="github account"
            options={mentorsList}
            handleInput={handleInput}
          />
          {mentorDataObj ? (
            <UserInfo
              mentorDataObj={mentorDataObj}
            />
          ) : <LoginButton />}
        </nav>
      </div>
    );
  }
}

NavBar.propTypes = {
  handleInput: PropTypes.func.isRequired,
  mentorDataObj: PropTypes.instanceOf(Object),
};

NavBar.defaultProps = {
  mentorDataObj: {},
};

export default NavBar;
