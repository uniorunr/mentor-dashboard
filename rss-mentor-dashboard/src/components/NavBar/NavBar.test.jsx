import React from 'react';
import renderer from 'react-test-renderer';
import * as firebase from 'firebase';
import NavBar from './NavBar';

const onAuthStateChanged = jest.fn();
const signInWithPopup = jest.fn(() => Promise.resolve('result of signInWithRedirect'));

jest.spyOn(firebase, 'initializeApp')
  .mockImplementation(() => ({
    auth: () => ({
      currentUser: {},
      signInWithPopup,
    }),
  }));

jest.spyOn(firebase, 'auth').mockImplementation(() => ({
  onAuthStateChanged,
  currentUser: {
    displayName: 'testDisplayName',
    email: 'test@test.com',
    emailVerified: true,
  },
}));

const handleInput = (mentorInput) => {
  this.setState({
    mentor: mentorInput,
  });
};

const database = {
  mentors: [{
    name: 'Test',
    surname: 'Test',
    fullName: 'Test',
    city: 'Test',
    count: 0,
    github: 'https://github.com/test',
    githubUsername: 'test',
    students: [{
      github: 'test',
      tasks: [{
        taskName: 'Test',
        normalizedTaskName: 'task',
        status: 'checked',
      },
      ],
    }],
  }],
  tasks: [{
    taskName: 'Test',
    normalizedTaskName: 'task',
    link: 'https://github.com/test',
    status: 'checked',
  }],
};

const mentorDataObj = {
  displayName: 'test',
  photoURL: 'test',
};

it('renders correctly', () => {
  const tree = renderer
    .create(<NavBar
      handleInput={handleInput}
      mentorDataObj={mentorDataObj}
      database={database}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
