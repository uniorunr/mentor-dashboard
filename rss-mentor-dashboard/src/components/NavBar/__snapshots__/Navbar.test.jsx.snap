// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="navbar-container"
>
  <div
    className="header-logo"
  >
    <h1>
      Director
    </h1>
  </div>
  <nav>
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/users"
        >
          Users
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
